### PR TITLE
yul Interpreter: Move memory size limit to interpreter state.

### LINKED ISSUE #6227: yulInterpreter: Configuration for fuzzing
## Description

The main configuration settings for the yulInterpreter that affect fuzzing are (defaults shown in brackets):
  - Max Size of memory used for emulating yul instructions (state memory=512MB)
  - Max number of traces until the interpreter times out (trace size=10000)

This needs a relook.

Also up for discussion is how to bail out from infinite loops in cases where the maxTraceSize won't help us.

Example
```
{for { let i := 1} i {} {}}
```

### Data

We currently set `maxTraceSize=10000` which is too large.

Data suggests the following

For non-crashing test inputs

|   Number of test inputs  |  Avg. trace size | Median trace size |
|---------------------------------|----------------------|-------------------------|
| 18112 | 5.4 | 3 |

For historical crashing test inputs

|   Number of test inputs  |  Avg. trace size | Median trace size |
|---------------------------------|----------------------|-------------------------|
| 22 | 4.1 | 3 |

For historical slow test inputs (inputs that slow down the fuzzing binary)

|   Number of test inputs  |  Avg. trace size | Median trace size |
|---------------------------------|----------------------|-------------------------|
| 80 | 463 | 131.5 |

For historical test inputs that increase memory allocation in the fuzzing binary

|   Number of test inputs  |  Avg. trace size | Median trace size |
|---------------------------------|----------------------|-------------------------|
| 8 | 226.5 | 226.5 |

As for maxMemorySize, I don't have data yet, but setting it to 512MB (current default) is too large and leads to out-of-memory. Also, I'm not sure if we want to strictly emulate memory loads and stores beyond a small window of say 1 KB. At the moment, programs such as the following can arbitrarily increase size of interpreter state memory.
```
{mstore(12321321, 0)}
```

Take aways:
  - The trace size distribution of non-crashing and crashing test inputs are somewhat similar although the latter has a very small sample size
  - The distribution of slow and out-of-memory test inputs is starkly different
  - Makes sense to reduce the maximum trace size threshold so that we don't spend time on slow/oom inputs
  - Reduce maxMemorySize to say 1 KB

### PR DESCRIPTION
**Pull request by bshastry**
### Description

(closes #6227 )

### CODE REVIEW

File: test/tools/ossfuzz/yulFuzzerCommon.cpp [hunk 1]
```diff
@@ -23,7 +23,7 @@ using namespace yul::test::yul_fuzzer;
 void yulFuzzerUtil::interpret(ostream& _os, shared_ptr<yul::Block> _ast)
 {
 	InterpreterState state;
-	state.maxTraceSize = 10000;
+	state.maxTraceSize = 75;
```
Comment (chriseth) [inline test/tools/ossfuzz/yulFuzzerCommon.cpp:26]: Isn't that a bit too short?

**Review by chriseth — APPROVED** [review]
75 might be a little short, but if you say it's fine then this can be merged.

File: test/tools/ossfuzz/yulFuzzerCommon.cpp (hunk 1 above)
Comment (bshastry) [reply test/tools/ossfuzz/yulFuzzerCommon.cpp:26]: That's what I felt in the beginning but looking at the data, almost all test inputs (including crash inducing ones) have a trace size under 50. The inputs that require larger traces are usually nested for loops with hundreds of inner loops.

Comment (chriseth) [reply test/tools/ossfuzz/yulFuzzerCommon.cpp:26]: Ok!

**Pull request merged**